<launch>
    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args= "-d $(find sobit_follower)/config/rviz/person_tracker.rviz" />
    <!-- ssd_node -->
    <include file="$(find sobit_follower)/launch/follow_me/object_detect.launch"/>
    <!-- person_aim_xtion_rotator -->
     <group ns = "person_aim_xtion_rotatorr">
        <node pkg="sobit_follower" type="sf_person_aim_xtion_rotator" name="person_aim_xtion_rotator">
            <rosparam>
                is_sobit_mini : false
                camera2person_height : 0.6
                tilt_angle_range : [ -1.0, -0.25 ]
            </rosparam>
        </node>
    </group>
    <!-- person_tracker -->
    <group ns = "person_tracker">
        <node pkg="sobit_follower" type="sf_person_tracker" name="person_tracker" output="screen">
            <remap from="/person_tracker/follow_position" to="/follow_position" />

            <rosparam>
                ssd_topic_name : /ssd_object_detect/objects_rect
                cloud_topic_name : /camera/depth/points
                laser_topic_name : /scan
                target_frame : base_footprint
                <!-- StandardPointCloudHandle -->
                search_radius : 0.25 
                <!-- LaserScanHandle -->
                laser_clustering_tolerance : 0.15
                laser_clustering_points_min_size : 1
                laser_clustering_points_max_size : 1000
                large_laser_clustering_tolerance : 0.15
                large_laser_clustering_points_min_size : 30
                large_laser_clustering_points_max_size : 1000
                neighborhood_search_radius : 0.1
                number_of_nearest_neighbors_radius : 2
                keep_organized : false
                <!-- PersonDetector -->
                detect_allowable_distance : 4.0
                tracking_allowable_distance : 0.3
                prediction_allowable_distance : 0.5
                <!-- FollowPositionHandle -->
                nearby_obstacle_distance : 0.3 
                <!-- PersonTracker -->
                cluster_size_large_obstacle : [0.5, 0.5] 
                track_update_time : 2.0
                predicted_execution_time : 2.0
                should_output_marker : true
            </rosparam>
        </node>
    </group>
</launch>